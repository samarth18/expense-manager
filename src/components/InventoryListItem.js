import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const InventoryListItem = ({
  id,
  description,
  netAmount,
  createdAt,
  quantity
}) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h3 className="list-item__title">{description}</h3>
      <div>
        <span className="list-item__sub-title">Quantity: {quantity}</span>
      </div>
      <span className="list-item__sub-title">
        {moment(createdAt).format("MMMM Do, YYYY")}
      </span>
    </div>
    <h3 className="list-item__data">{numeral(netAmount).format("$0,0.00")}</h3>
  </Link>
);

export default InventoryListItem;
