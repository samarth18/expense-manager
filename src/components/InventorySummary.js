import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import numeral from "numeral";
import selectItems from "../selectors/items";
import selectItemsTotal from "../selectors/items-total";

export const InventorySummary = ({ itemCount, itemsTotal }) => {
  const itemWord = itemCount === 1 ? "item" : "items";
  const formattedItemsTotal = numeral(itemsTotal).format("$0,0.00");

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">
          Displaying <span>{itemCount}</span> {itemWord} with a total of{" "}
          <span>{formattedItemsTotal}</span>
        </h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">
            <span>Add Item</span>
          </Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleItems = selectItems(state.items, state.filters);

  return {
    itemCount: visibleItems.length,
    itemsTotal: selectItemsTotal(visibleItems)
  };
};

export default connect(mapStateToProps)(InventorySummary);
