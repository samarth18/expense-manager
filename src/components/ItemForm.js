import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

export default class ItemForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.item ? props.item.description : "",
      note: props.item ? props.item.note : "",
      quantity: props.item ? props.item.quantity.toString() : "",
      cost: props.item ? (props.item.cost / 100).toString() : "",
      netAmount: props.item
        ? props.item.quantity * (props.item.cost / 100)
        : "",
      createdAt: props.item ? moment(props.item.createdAt) : moment(),
      calendarFocused: false,
      error: ""
    };
  }
  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onQuantityChange = e => {
    const quantity = e.target.value;

    if (!quantity || quantity.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ quantity }));
    }
  };
  onAmountChange = e => {
    const cost = e.target.value;

    if (!cost || cost.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ cost }));
    }
  };
  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = e => {
    e.preventDefault();

    if (!this.state.description || !this.state.cost || !this.state.quantity) {
      this.setState(() => ({
        error: "Please provide description, quantity and cost."
      }));
    } else {
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        description: this.state.description,
        quantity: parseFloat(this.state.quantity, 10),
        cost: parseFloat(this.state.cost, 10) * 100,
        netAmount:
          parseFloat(this.state.quantity, 10) * parseFloat(this.state.cost, 10),
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <input
          type="text"
          placeholder="Description"
          autoFocus
          className="text-input"
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />
        <input
          type="text"
          placeholder="Quantity"
          className="text-input"
          value={this.state.quantity}
          onChange={this.onQuantityChange}
        />
        <input
          type="text"
          placeholder="Cost"
          className="text-input"
          value={this.state.cost}
          onChange={this.onAmountChange}
        />
        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onDateChange}
          focused={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
        <textarea
          placeholder="Add a note for your item (optional)"
          className="textarea"
          value={this.state.note}
          onChange={this.onNoteChange}
        />
        <div>
          <button className="button">
            <span>Save Item</span>
          </button>
        </div>
      </form>
    );
  }
}
