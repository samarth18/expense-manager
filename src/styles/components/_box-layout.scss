.box-layout {
  align-items: center;
  background: url("/images/nyc1.jpg");
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(white, 0.15);
  border-radius: 10px;
  padding: $l-size $m-size;
  text-align: center;
  width: 25rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
