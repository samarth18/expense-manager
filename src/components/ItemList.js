import React from "react";
import { connect } from "react-redux";
import InventoryListItem from "./InventoryListItem";
import selectItems from "../selectors/items";

export const ItemList = props => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Items</div>
      <div className="show-for-desktop">Item</div>
      <div className="show-for-desktop">Net Amount</div>
    </div>
    <div className="list-body">
      {props.items.length === 0 ? (
        <div className="list-item list-item--message">
          <span>No items</span>
        </div>
      ) : (
        props.items.map(item => {
          return <InventoryListItem key={item.id} {...item} />;
        })
      )}
    </div>
  </div>
);

const mapStateToProps = state => {
  return {
    items: selectItems(state.items, state.filters)
  };
};

export default connect(mapStateToProps)(ItemList);
