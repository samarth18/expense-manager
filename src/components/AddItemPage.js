import React from "react";
import { connect } from "react-redux";
import ItemForm from "./ItemForm";
import { startAddItem } from "../actions/items";

export class AddItemPage extends React.Component {
  onSubmit = item => {
    this.props.startAddItem(item);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Item</h1>
          </div>
        </div>
        <div className="content-container">
          <ItemForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddItem: item => dispatch(startAddItem(item))
});

export default connect(
  undefined,
  mapDispatchToProps
)(AddItemPage);
