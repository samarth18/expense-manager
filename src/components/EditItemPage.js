import React from "react";
import { connect } from "react-redux";
import ItemForm from "./ItemForm";
import { startEditItem, startRemoveItem } from "../actions/items";

export class EditItemPage extends React.Component {
  onSubmit = item => {
    this.props.startEditItem(this.props.item.id, item);
    this.props.history.push("/");
  };
  onRemove = () => {
    this.props.startRemoveItem({ id: this.props.item.id });
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Item</h1>
          </div>
        </div>
        <div className="content-container">
          <ItemForm item={this.props.item} onSubmit={this.onSubmit} />
          <button className="button button--secondary" onClick={this.onRemove}>
            <span>Remove Item</span>
          </button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  item: state.items.find(item => item.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditItem: (id, item) => dispatch(startEditItem(id, item)),
  startRemoveItem: data => dispatch(startRemoveItem(data))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditItemPage);
