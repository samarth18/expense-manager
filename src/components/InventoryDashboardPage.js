import React from "react";
import ItemList from "./ItemList";
import ItemListFilters from "./ItemListFilters";
import InventorySummary from "./InventorySummary";

const InventoryDashboardPage = () => (
  <div>
    <InventorySummary />
    <ItemListFilters />
    <ItemList />
  </div>
);

export default InventoryDashboardPage;
