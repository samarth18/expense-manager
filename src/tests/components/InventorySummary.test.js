import React from "react";
import { shallow } from "enzyme";
import { InventorySummary } from "../../components/InventorySummary";

test("should correctly render InventorySummary with 1 item", () => {
  const wrapper = shallow(<InventorySummary itemCount={1} itemsTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render InventorySummary with multiple items", () => {
  const wrapper = shallow(
    <InventorySummary itemCount={23} itemsTotal={23512340987} />
  );
  expect(wrapper).toMatchSnapshot();
});
