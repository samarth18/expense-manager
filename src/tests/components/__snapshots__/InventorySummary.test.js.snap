// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render InventorySummary with 1 item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InventorySummary
    itemCount={1}
    itemsTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
>
        <h1
                className="page-header__title"
        >
                Displaying 
                <span>
                        1
                </span>
                 
                item
                 with a total of
                 
                <span>
                        $235.00
                </span>
        </h1>
        <div
                className="page-header__actions"
        >
                <Link
                        className="button"
                        replace={false}
                        to="/create"
                >
                        <span>
                                Add Item
                        </span>
                </Link>
        </div>
</div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
>
            Displaying 
            <span>
                        1
            </span>
             
            item
             with a total of
             
            <span>
                        $235.00
            </span>
</h1>,
          <div
            className="page-header__actions"
>
            <Link
                        className="button"
                        replace={false}
                        to="/create"
            >
                        <span>
                                    Add Item
                        </span>
            </Link>
</div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Displaying ",
              <span>
                1
</span>,
              " ",
              "item",
              " with a total of",
              " ",
              <span>
                $235.00
</span>,
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Displaying ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
            " ",
            "item",
            " with a total of",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$235.00",
              },
              "ref": null,
              "rendered": "$235.00",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/create"
>
              <span>
                            Add Item
              </span>
</Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                Add Item
</span>,
              "className": "button",
              "replace": false,
              "to": "/create",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Item",
              },
              "ref": null,
              "rendered": "Add Item",
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
>
          <h1
                    className="page-header__title"
          >
                    Displaying 
                    <span>
                              1
                    </span>
                     
                    item
                     with a total of
                     
                    <span>
                              $235.00
                    </span>
          </h1>
          <div
                    className="page-header__actions"
          >
                    <Link
                              className="button"
                              replace={false}
                              to="/create"
                    >
                              <span>
                                        Add Item
                              </span>
                    </Link>
          </div>
</div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
>
              Displaying 
              <span>
                            1
              </span>
               
              item
               with a total of
               
              <span>
                            $235.00
              </span>
</h1>,
            <div
              className="page-header__actions"
>
              <Link
                            className="button"
                            replace={false}
                            to="/create"
              >
                            <span>
                                          Add Item
                            </span>
              </Link>
</div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Displaying ",
                <span>
                  1
</span>,
                " ",
                "item",
                " with a total of",
                " ",
                <span>
                  $235.00
</span>,
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Displaying ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
              " ",
              "item",
              " with a total of",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$235.00",
                },
                "ref": null,
                "rendered": "$235.00",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/create"
>
                <span>
                                Add Item
                </span>
</Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  Add Item
</span>,
                "className": "button",
                "replace": false,
                "to": "/create",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Item",
                },
                "ref": null,
                "rendered": "Add Item",
                "type": "span",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should correctly render InventorySummary with multiple items 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InventorySummary
    itemCount={23}
    itemsTotal={23512340987}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
>
        <h1
                className="page-header__title"
        >
                Displaying 
                <span>
                        23
                </span>
                 
                items
                 with a total of
                 
                <span>
                        $23,512,340,987.00
                </span>
        </h1>
        <div
                className="page-header__actions"
        >
                <Link
                        className="button"
                        replace={false}
                        to="/create"
                >
                        <span>
                                Add Item
                        </span>
                </Link>
        </div>
</div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
>
            Displaying 
            <span>
                        23
            </span>
             
            items
             with a total of
             
            <span>
                        $23,512,340,987.00
            </span>
</h1>,
          <div
            className="page-header__actions"
>
            <Link
                        className="button"
                        replace={false}
                        to="/create"
            >
                        <span>
                                    Add Item
                        </span>
            </Link>
</div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Displaying ",
              <span>
                23
</span>,
              " ",
              "items",
              " with a total of",
              " ",
              <span>
                $23,512,340,987.00
</span>,
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Displaying ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 23,
              },
              "ref": null,
              "rendered": 23,
              "type": "span",
            },
            " ",
            "items",
            " with a total of",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$23,512,340,987.00",
              },
              "ref": null,
              "rendered": "$23,512,340,987.00",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/create"
>
              <span>
                            Add Item
              </span>
</Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                Add Item
</span>,
              "className": "button",
              "replace": false,
              "to": "/create",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Item",
              },
              "ref": null,
              "rendered": "Add Item",
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
>
          <h1
                    className="page-header__title"
          >
                    Displaying 
                    <span>
                              23
                    </span>
                     
                    items
                     with a total of
                     
                    <span>
                              $23,512,340,987.00
                    </span>
          </h1>
          <div
                    className="page-header__actions"
          >
                    <Link
                              className="button"
                              replace={false}
                              to="/create"
                    >
                              <span>
                                        Add Item
                              </span>
                    </Link>
          </div>
</div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
>
              Displaying 
              <span>
                            23
              </span>
               
              items
               with a total of
               
              <span>
                            $23,512,340,987.00
              </span>
</h1>,
            <div
              className="page-header__actions"
>
              <Link
                            className="button"
                            replace={false}
                            to="/create"
              >
                            <span>
                                          Add Item
                            </span>
              </Link>
</div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Displaying ",
                <span>
                  23
</span>,
                " ",
                "items",
                " with a total of",
                " ",
                <span>
                  $23,512,340,987.00
</span>,
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Displaying ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 23,
                },
                "ref": null,
                "rendered": 23,
                "type": "span",
              },
              " ",
              "items",
              " with a total of",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$23,512,340,987.00",
                },
                "ref": null,
                "rendered": "$23,512,340,987.00",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/create"
>
                <span>
                                Add Item
                </span>
</Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  Add Item
</span>,
                "className": "button",
                "replace": false,
                "to": "/create",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Item",
                },
                "ref": null,
                "rendered": "Add Item",
                "type": "span",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
